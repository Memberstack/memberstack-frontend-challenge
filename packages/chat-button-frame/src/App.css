.chat-icon {
  border-radius: 50%;
  display: inline-flex;
  padding: 10px;
  background-color: #cfeaf7;
  border: 1px solid rgba(81, 92, 93, 0.1);
  position: absolute;
  bottom: 0;
  right: 0;
}

.chat-icon-image {
  height: 30px;
}
