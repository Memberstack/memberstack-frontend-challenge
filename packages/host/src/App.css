.chatbot-iframe {
  position: absolute;
  bottom: 20px;
  right: 20px;
}

iframe.chatbot-iframe > .chat-icon {
  background-color: black !important;
}
