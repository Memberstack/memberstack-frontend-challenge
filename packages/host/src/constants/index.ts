export const INITIAL_COLOR = '#e66465';