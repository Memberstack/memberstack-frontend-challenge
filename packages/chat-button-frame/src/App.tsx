import './App.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCommentDots } from '@fortawesome/free-solid-svg-icons';
import {useState, useEffect} from 'react';

function App() {

  const [color, setColor] = useState("")

  useEffect(() => {
    const value = document.URL.split("color=")[1];
    setColor((prevValue) => value);
    //---------- *Improvements[2] --------------
  });

  return (
    <div className="App">
      <div className="chat-icon">
        <FontAwesomeIcon icon={faCommentDots} size='2x' color={color}/>
        {/* <img src={logo} className="chat-icon-image" alt="logo" /> */}
      </div>
    </div>
  );
}

export default App;
