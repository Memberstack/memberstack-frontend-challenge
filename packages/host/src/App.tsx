import ColorPicker from "./components/ColorPicker";
import ChatBotIFrame from "./components/ChatBotIFrame";
import { useLocalStorage } from "./hooks/useLocalStorage";
import { INITIAL_COLOR } from "./constants";
import { Container, 
  Main, 
  IFrameContainer, 
  ContentContainer, 
  Header, 
  Footer, 
  Nav, 
  Link,
  Text
} from "./components";

const App = () => {

  const {item: color, setItem: setColor} = useLocalStorage(INITIAL_COLOR, 'color')

  const handleColorChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setColor(event.target.value)
  };

  return (
    <Container>
      <Main>
        <IFrameContainer>
          <ChatBotIFrame value={color}/>
        </IFrameContainer>
        <ContentContainer>
          <Header>ColorPicker Demo Challenge</Header>
          <Text>Select a color from the ColorPicker and watch the chatbox icon change</Text>
          <ColorPicker
            value={color}
            onChange={handleColorChange}
          />
        </ContentContainer>
      </Main>
      <Footer>
        <Nav>
          <Link href='https://github.com/clinton-slice/memberstack-frontend-challenge'>
            Check the code out on GitHub
          </Link>
        </Nav>
      </Footer>
    </Container>
  );
}

export default App;

