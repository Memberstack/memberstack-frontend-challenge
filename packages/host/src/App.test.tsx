import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';
import { INITIAL_COLOR } from './constants';

describe('App', ()=>{
  describe('sanity', ()=>{
    it('matches snapshot', ()=>{
      const {container} = render(<App/>)
      expect(container.firstChild).toMatchSnapshot();
    })
    it('renders Check the code out on GitHub link', () => {
      render(<App />);
      const linkElement = screen.getByText(/Check the code out on GitHub/i);
      expect(linkElement).toBeInTheDocument();
    });

    it('input has correct initial value', () => {
      render(<App />);
      const colorPickerInput = screen.getByTestId(/color-test/i);
      expect(colorPickerInput).toHaveValue(INITIAL_COLOR);
    });
  })

  describe('ColorPickerInput value changes onChange', ()=>{
    it('color value changes', ()=>{
      render(<App/>)
      const colorPickerInput = screen.getByTestId(/color-test/i);
      fireEvent.change(colorPickerInput, {target: {value: '#292d3e'}})
      expect(colorPickerInput).toHaveValue('#292d3e');
    })
  })
});