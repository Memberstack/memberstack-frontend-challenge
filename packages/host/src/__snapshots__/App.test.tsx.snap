// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App sanity matches snapshot 1`] = `
<div
  class="sc-gtsrHT ismbld"
>
  <main
    class="sc-dlnjwi gxTIdd"
  >
    <div
      class="sc-hKFxyN huoWko"
    >
      <iframe
        class="chatbot-iframe"
        frameborder="0"
        src="http://localhost:3002/"
        title="Chat bot"
      />
    </div>
    <div
      class="sc-eCApnc hgCFrt"
    >
      <h1
        class="sc-jSFjdj gxeEdc"
      >
        ColorPicker Demo Challenge
      </h1>
      <p
        class="sc-gKAaRy kdDXpC"
      >
        Select a color from the ColorPicker and watch the chatbox icon change
      </p>
      <input
        class="sc-bdnxRM cAjHbq"
        data-testid="color-test"
        type="color"
        value="#e66465"
      />
    </div>
  </main>
  <footer
    class="sc-iCoGMd cGOaZn"
  >
    <nav
      class="sc-fujyAs bvOCsI"
    >
      <a
        class="sc-pNWdM bXhOie"
        href="https://github.com/clinton-slice/memberstack-frontend-challenge"
      >
        Check the code out on GitHub
      </a>
    </nav>
  </footer>
</div>
`;
